lib/iex/lib/iex/autocomplete.ex

```diff
@@ -643,4 +647,92 @@ defmodule IEx.Autocomplete do
   defp extract_from_ast(_ast_node, _acc) do
     :error
   end
+
+  defp path_fragment(expr), do: collect_path_fragment(expr, [])
+
+  # collects the fragment and store it on `acc`, calls `return_path_fragment` 
+  # when it finds a `"`, discards `acc` otherwise
+  defp collect_path_fragment([], _acc), do: []
+  defp collect_path_fragment([?", ?\\ | t], acc), do: collect_path_fragment(t, [?\\, ?" | acc])
+  defp collect_path_fragment([?{, ?# | _rest], _acc), do: []
+  defp collect_path_fragment([?" | t], acc), do: return_path_fragment(t, acc)
+  defp collect_path_fragment([h | t], acc), do: collect_path_fragment(t, [h | acc])
+
+  # scans the rest of the `expr`, returns `acc` if no other `"` is found, 
+  # fallback to `discard_path_fragment` otherwise
+  defp return_path_fragment([], acc), do: acc
+  defp return_path_fragment([?", ?\\ | t], acc), do: return_path_fragment(t, [?\\, ?" | acc])
+  defp return_path_fragment([?" | t], acc), do: discard_path_fragment(t, acc)
+  defp return_path_fragment([_ | t], acc), do: return_path_fragment(t, acc)
+
+  # scans the rest of the `expr`, discards `acc` if no other `"` is found, 
+  # fallback to `return_path_fragment` otherwise
+  defp discard_path_fragment([], _acc), do: []
+  defp discard_path_fragment([?", ?\\ | t], acc), do: discard_path_fragment(t, [?\\, ?" | acc])
+  defp discard_path_fragment([?" | t], acc), do: return_path_fragment(t, acc)
+  defp discard_path_fragment([_ | t], acc), do: discard_path_fragment(t, acc)
+
+  defp expand_path(path_fragment) do
+    path_fragment = to_string(path_fragment)
+    possible_paths = find_possible_paths(path_fragment)
+    expand_path(path_fragment, possible_paths)
+  end
+
+  defp expand_path(path_fragment, possible_paths) do
+    expansions =
+      Enum.map(possible_paths, fn {path, dir?} ->
+        {completion_part(path_fragment, path, dir?), to_charlist(Path.basename(path))}
+      end)
+
+    case expansions do
+      [] ->
+        {:no, '', []}
+
+      [{unique, _}] ->
+        {:yes, unique, []}
+
+      list ->
+        {completions, filenames} = Enum.unzip(list)
+        hint = Enum.reduce(completions, &common_prefix/2)
```

Comment:
See if you can use http://erlang.org/doc/man/binary.html#longest_common_prefix-1